app/Policies/UserPolicy.php

```diff
@@ -156,14 +176,33 @@ public function updatePermission(User $user, User $target, Role $role, Workshop|
             return $user->roleWorkshops()->contains($object->id);
         }
 
+        if ($role->name == Role::WORKSHOP_LEADER) {
+            return $user->hasRole(Role::SECRETARY) || $user->hasRole(Role::DIRECTOR);
+        }
+
+        if ($role->name == Role::WORKSHOP_ADMINISTRATOR) {
+            return $user->hasRoleBase(Role::WORKSHOP_LEADER) || $user->hasRole(Role::STUDENT_COUNCIL_SECRETARY) || $user->hasRole(Role::SECRETARY) || $user->hasRole(Role::STUDENT_COUNCIL, Role::SCIENCE_VICE_PRESIDENT);
```

Comment:
Only the corresponding workshop leader should add the workshop administrator role. I have similar functionalities somewhere with a $user->roleWorkshops call or sth similar